src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -186,6 +188,120 @@ private with sharing class RD2_OpportunityService_TEST {
+
+        Id defaultRecordTypeId = UTIL_RecordTypes_TEST.getDefaultRecordTypeId(Opportunity.sObjectType);
+
+        System.assertNotEquals(null, installmentOpp, 'An opportunity should be created for each recurring donation record');
+        System.assertEquals(defaultRecordTypeId, installmentOpp.RecordTypeId,
+            'The record type should be the default opp record type when NPSP default record type is null');
+    }
+
+    /***
+    * @description The installment opportunity should keep the original record type when the NPSP opp default record type
+    * is updated
+    */
+    @isTest
+    private static void shouldNotUpdateOppRecordTypeWhenRecordTypeSettingChange() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        createRdSettingsWithFieldMap(null);
+        npe03__Recurring_Donation__c rd = createRecurringDonation();
+
+        Opportunity installmentOpp = new TEST_OpportunityBuilder()
+            .withAmount(100)
+            .withAccount(getContact().AccountId)
+            .withClosedWonStage()
+            .withCloseDate(Date.today())
+            .withRecurringDonation(rd.Id)
+            .withCampaign(rd.npe03__Recurring_Donation_Campaign__c)
+            .build();
+        insert installmentOpp;
+
+        Test.startTest();
+        Id rdSettingRecordTypeId = UTIL_RecordTypes_TEST.getNonDefaultRecordTypeId(Opportunity.sObjectType);
+
+        Map<String, Object> rdSettingFieldMap = new Map<String, Object>{
+            'npe03__Record_Type__c' => rdSettingRecordTypeId
+        };
+        createRdSettingsWithFieldMap(rdSettingFieldMap);
+        
+        Test.stopTest();
+
+        installmentOpp = getOpportunities()[0];   
+
+        System.assertNotEquals(rdSettingRecordTypeId, installmentOpp.RecordTypeId,
+            'The record type should not be updated to NPSP default record type if the opp is already created before the update');
+    }
+
+    /***
+    * @description Error__c records should be created and opportunity records should not be inserted 
+    *   when creating a RD if the NPSP Opportunity record type is invalid
+    */
+    @isTest
+    private static void shouldLogErrorAndBlockOppCreationWhenOppDefaultRecordTypeIsInvalid() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Map<String, Object> rdSettingFieldMap = new Map<String, Object>{
+            'npe03__Record_Type__c' => RECORD_TYPE_MOCK_ID
+        };
+        createRdSettingsWithFieldMap(rdSettingFieldMap);
+
+        Test.startTest();
+        createRecurringDonations(2);
+        Test.stopTest();
+
+        System.assertEquals(0, getOpportunities().size(), 
+            'No installment opportunity should be created when the default record type on the setting is invalid');
+
+        List<Error__c> errors = errorGateway.getRecords();
+
+        System.assertEquals(2, errors.size(), '2 Error records should be created for each installment opp');
```

Comment:
Suggestion: instead of specifying details in the assert message, we can have generic message, so when/if `assert` arguments change, the message is still valid: 
`System.assertEquals(rdSize, errors.size(), 'Number of error should match');`

WHERE
- `rdSize` is number of RD records (to be used in line 290 as well).